src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let separatorString = separator;
+
+    if (separator === null) {
+      separatorString = 'null';
```

Comment:
that will work as you want anyway if you try to `result += separatorString`, so you can remove this condition